File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string result = "";
        for (string& s: strs) {
            string temp = to_string(s.size()) + '#' + s;
            result += temp;
        }
        return result;
    }

    vector<string> decode(string s) {
        vector<string> result;

        int i = 0;
        while (i < s.size()) {
            int j = i;
            while (s[j] != '#') {
                j++;
            }
            //Start String : 4#neet
            //e0: 4, e1: #, start:3-5
            //First run; i = 0; j = 1;
            int length = stoi(s.substr(i, j));
            string str = s.substr(j + 1, length);
            result.push_back(str);
            i = j + 1 + length;
        }
        return result;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = {}

        for i in strs:
            temp = "".join(sorted(i))
            if temp in mp:
                mp[temp].append(i)
            else:
                mp[temp] = [i]
        return list(mp.values())